model/job.go

```diff
@@ -225,6 +234,124 @@ func (j *Job) loadJobSpec() (err error) {
 	return nil
 }
 
+// MergeSpec is used to merge temporary spec patches into each job. otherJob should only be
+// the hcf/patch-properties job.  The code assumes package and property objects are immutable,
+// as they're now being shared across jobs. Also, the results are undefined if specified packages
+// or properties already exist in the destination job.
```

Comment:
But we _do_ have things like the autoscaler where the property already exists in some jobs but not the others, right?